libs/db/cpp/db.hpp

```diff
@@ -0,0 +1,3728 @@
+#ifndef ELODIN_DB_H
+#define ELODIN_DB_H
+
+#include <stdio.h>
+#include <span>
+#include <variant>
+#include <type_traits>
+#include <vector>
+#include <optional>
+
+
+#if __has_include("postcard.h")
+#include "postcard.h"
+#endif
+
+
+#ifndef POSTCARD_H
```

Comment:
Could we avoid duplicating the contents of `postcard.h`? It's a bit annoying to maintain. I think requiring end users to import 2 header files is fine.

Reply:
This is an out-generated file that automatically imports postcard.h. So you just run `elodin-db gen-cpp` and it will auto generate